internal/daemon/controller/testing.go

```diff
@@ -632,6 +632,15 @@ func TestControllerConfig(t testing.TB, ctx context.Context, tc *TestController,
 	if opts.Config.Controller.Name == "" {
 		require.NoError(t, opts.Config.Controller.InitNameIfEmpty(ctxTest))
 	}
+
+	if opts.SchedulerRunJobInterval == 0 {
+		if t, ok := t.(*testing.T); ok {
+			if deadline, ok := t.Deadline(); ok {
+				half := int64(time.Until(deadline) / 2)
```

Comment:
😅 it was strictly less than, I take it? Mind adding a comment here explaining why this is this way?

Reply:
Great suggestion! I added doc to the option itself to make it more visible. Please let me know what you think.